barretenberg/cpp/src/barretenberg/dsl/acir_format/honk_recursion_constraint.test.cpp

```diff
@@ -1,440 +1,489 @@
 #include "honk_recursion_constraint.hpp"
 #include "acir_format.hpp"
-#include "acir_format_mocks.hpp"
 #include "barretenberg/chonk/mock_circuit_producer.hpp"
 #include "barretenberg/dsl/acir_format/gate_count_constants.hpp"
+#include "barretenberg/dsl/acir_format/test_class_predicate.hpp"
 #include "barretenberg/dsl/acir_format/utils.hpp"
 #include "barretenberg/dsl/acir_format/witness_constant.hpp"
 #include "barretenberg/numeric/uint256/uint256.hpp"
 #include "barretenberg/special_public_inputs/special_public_inputs.hpp"
 #include "barretenberg/ultra_honk/prover_instance.hpp"
 #include "barretenberg/ultra_honk/ultra_prover.hpp"
 #include "barretenberg/ultra_honk/ultra_verifier.hpp"
-#include "proof_surgeon.hpp"
 
 #include <gtest/gtest.h>
 #include <vector>
 
 using namespace acir_format;
 using namespace bb;
 
-template <typename RecursiveFlavor> class AcirHonkRecursionConstraint : public ::testing::Test {
+/**
+ * These functions are used to generate the constraints for testing Honk recursive verification.
+ *
+ * We wish to test the following scenarios:
+ * 1. Vanilla recursion
```

Comment:
I'd avoid the term "vanilla" here. Historically we've used it to mean recursion in which we directly emulate non-native field arithmetic (i.e. non-Mega) but even that's usage isnt standard/useful to someone reading externally. Maybe just "a single recursion"?